src/backends/backend.js

```diff
@@ -261,7 +266,43 @@ class Backend {
     .then(this.entryWithFormat(collection, slug));
   }
 
-  persistEntry(config, collection, entryDraft, MediaFiles, integrations, options = {}) {
+  async checkOverwrite(collection, slug, path, entryData) {
+    const identifierKey = getIdentifierKey(entryData);
+    const identifierField = collection.get('fields').find(field => {
+      return field.get('name') === identifierKey;
+    });
+    const identifierLabel = identifierField.get('label');
+    const errorMessage = `\
+Duplicate filename found. Please ensure the ${identifierLabel} field is unique from other entries \
+in the ${collection.get('name')} collection.\
+`;
+
+    const existingEntry = await this.unpublishedEntry(collection, slug).catch(error => {
+      if (error instanceof EditorialWorkflowError && error.notUnderEditorialWorkflow) {
+        return Promise.resolve(false);
+      }
+      return Promise.reject(error);
+    });
+
+    if (existingEntry) {
+      throw new Error(errorMessage);
+    }
+
+    const publishedEntry = await this.implementation.getEntry(collection, slug, path)
+      .then(({ data }) => data)
+      .catch(error => {
+        if (error.message === 'Not Found') {
+          return Promise.resolve(false);
+        }
+        return Promise.reject(error);
+      });
```

Comment:
Note for reviewers: the `catch` method being added is the only new change in this PR compared to the original PR.